libs/mpi/include/hpx/mpi/mpi_future.hpp

```diff
@@ -0,0 +1,515 @@
+//  Copyright (c) 2019 John Biddiscombe
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_MPI_FUTURE_HPP
+#define HPX_MPI_FUTURE_HPP
+
+#include <mpi.h>
+//
+#include <array>
+#include <cstdio>
+#include <exception>
+#include <iostream>
+#include <list>
+#include <mutex>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include <hpx/concurrency/concurrentqueue.hpp>
+#include <hpx/memory/intrusive_ptr.hpp>
+#include <hpx/resource_partitioner/partitioner.hpp>
+#include <hpx/runtime/thread_pool_helpers.hpp>
+#include <hpx/threading_base/print.hpp>
+#include <hpx/lcos/detail/future_data.hpp>
+
+// The preferred mode of operation uses a std::vector and MPI_Testany
+// but the use of a list and individual tests is supported for legacy
+// debug reasons, uncomment one of these #defines to choose which is used
+//#define HPX_MPI_LIST_STORAGE
+#define HPX_MPI_VECTOR_STORAGE
+
+namespace hpx { namespace mpi {
```

Comment:
My preference would be `hpx::mpi::experimental` (i.e. `experimental` as the leaf namespace, just like `detail`). (Note: if the module name is `mpi_support` the namespace should probably be `hpx::mpi_support`).

Reply:
`hpx::mpi::experimental` is fine

Reply:
I will do some more tidying up of this PR today